src/RobotsTxt.php

```diff
@@ -126,36 +126,49 @@ protected function getDisallowsPerUserAgent(string $content): array
 
         $disallowsPerUserAgent = [];
 
-        $currentUserAgent = null;
+        $currentUserAgents = [];
+
+        $treatAllowDisallowLine = false;
 
         foreach ($lines as $line) {
-            if ($this->isCommentLine($line)) {
+            if ($this->isCommentOrEmptyLine($line)) {
                 continue;
             }
 
             if ($this->isUserAgentLine($line)) {
+                if ($treatAllowDisallowLine) {
+                    $treatAllowDisallowLine = false;
+                    $currentUserAgents = [];
+                }
                 $disallowsPerUserAgent[$this->parseUserAgent($line)] = [];
 
-                $currentUserAgent = &$disallowsPerUserAgent[$this->parseUserAgent($line)];
+                $currentUserAgents[] = &$disallowsPerUserAgent[$this->parseUserAgent($line)];
 
                 continue;
             }
 
-            if ($currentUserAgent === null) {
+            if ($this->isDisallowLine($line)) {
+                $treatAllowDisallowLine = true;
+            } elseif ($this->isAllowLine($line)) {
```

Comment:
In general I don't like `else` statements. Can this be refactored?

Reply:
refator done